Version 1.6.2 (#3304)
Kotlin/kotlinx.coroutines: CHANGES.md

# Change log for kotlinx.coroutines

## Version 1.6.2

> * Fixed a bug with `ThreadLocalElement` not being correctly updated when the most outer `suspend` function was started directly without `kotlinx.coroutines` (#2930).

Reviewer: ```suggestion
* Fixed a bug with `ThreadLocalElement` not being correctly updated when the outermost `suspend` function was called directly without `kotlinx.coroutines` (#2930).
```

> * Fixed multiple data races that might have been affecting `Mutex` and `runBlocking` event loop (#3250, #3251).

Reviewer: The data race in `Mutex` was benign (that is, shouldn't affect anything in practice), maybe we should highlight it.

Author: I'm not sure it really matters here

Reviewer: I just think that reading the changelog and seeing that `Mutex` had a data race is pretty scary, given that its whole purpose is to guard against them.

> * Fixed multiple data races that might have been affecting `runBlocking` event loop and benign data race in `Mutex` (#3250, #3251).

Reviewer: ```suggestion
* Fixed multiple data races: one that might have been affecting `runBlocking` event loop, and a benign data race in `Mutex` (#3250, #3251).
```

> * Obsolete `TestCoroutineContext` is removed along with JPMS split-package between `kotlinx-coroutines-core` and `kotlinx-coroutines-test` (#3218).

Reviewer: ```suggestion
* Obsolete `TestCoroutineContext` is removed, which fixes the `kotlinx-coroutines-test` JPMS package being split between `kotlinx-coroutines-core` and `kotlinx-coroutines-test` (#3218).
```

> * Updated ProGuard rules that help further shrink the size of the resulting DEX file with coroutines (#3263).

Reviewer: ```suggestion
* Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3263).
```

Reviewer: Also, attribution.

> * Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3263). Thanks @agrieve!

Reviewer: ```suggestion
* Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3111). Thanks, @agrieve!
```
We usually reference the issues, right?

Author: I was thinking the PR is more insightful. I've left both references

* Atomicfu is updated to `0.17.2`, which includes a more efficient and robust JS IR transformer (#3255).

> * Kotlin is updated to `1.6.21`, Gradle version is updated to `7.4.2` (#3281). Thanks @wojtek-kalicinski!

Reviewer: ```suggestion
* Kotlin is updated to `1.6.21`, Gradle version is updated to `7.4.2` (#3281). Thanks, @wojtek-kalicinski!
```
If we're treating the changelog as a somewhat official document.

Author: Done
